fix(release): Some known HTTP::ConnectionError issues resolved in http 5.1.x

jrafanie opened the pull request:
* Note, there are some breaking changes in version 5.0.0:
https://github.com/httprb/http/blob/main/CHANGES.md#500-2021-05-12
* Update ibm_cloud_sdk_core to 1.2.0 for http 5.1.x support
* Bump rake dependency to support http 5.1.x
* http 5.1.1 requires ruby 2.6+, test with more modern rubies

~~NOTE 2, this PR will depend on https://github.com/IBM/ruby-sdk-core/pull/40 being merged, released, and bumping the `ibm_cloud_sdk_core` dependency in the various gemspecs in this repository.~~

Similar change as https://github.com/IBM-Cloud/ibm-cloud-sdk-ruby/pull/73

agrare commented:
@jrafanie they use semantic-release here so I think we need `fix(release)` in one of the commit messages https://github.com/semantic-release/semantic-release but I'm not sure if `Fix some ...` would count

jrafanie commented:
> @jrafanie they use semantic-release here so I think we need `fix(release)` in one of the commit messages https://github.com/semantic-release/semantic-release but I'm not sure if `Fix some ...` would count

Ok, fixed and squashed.

agrare commented:
Okay here goes nothing :crossed_fingers:

ibm-vpc commented:
:tada: This PR is included in version 0.6.1 :tada:

The release is available on [GitHub release](https://github.com/IBM/vpc-ruby-sdk/releases/tag/0.6.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
